src/libs/data-table-versions.js

```diff
@@ -1,44 +1,77 @@
 import { addMinutes, format } from 'date-fns/fp'
+import JSZip from 'jszip'
 import _ from 'lodash/fp'
 import { useState } from 'react'
-import { notifyDataImportProgress, parseGsUri } from 'src/components/data/data-utils'
+import { parseGsUri } from 'src/components/data/data-utils'
 import { Ajax } from 'src/libs/ajax'
 import { getUser } from 'src/libs/auth'
 import Events, { extractWorkspaceDetails } from 'src/libs/events'
 import { notify } from 'src/libs/notifications'
 import { useCancellation } from 'src/libs/react-utils'
-import { asyncImportJobStore } from 'src/libs/state'
 import * as Utils from 'src/libs/utils'
 
 
 export const dataTableVersionsPathRoot = '.data-table-versions'
 
-export const saveDataTableVersion = async (workspace, entityType, { description = null } = {}) => {
+const getAllEntities = async (workspace, entityType) => {
+  const { workspace: { namespace, name } } = workspace
+
+  const pageSize = 100_000
+  const firstPageResponse = await Ajax().Workspaces.workspace(namespace, name).paginatedEntitiesOfType(entityType, { page: 1, pageSize })
+
+  let entities = firstPageResponse.results
+  const numPages = firstPageResponse.resultMetadata.filteredPageCount
+
+  for (let page = 2; page <= numPages; page += 1) {
+    const pageResponse = await Ajax().Workspaces.workspace(namespace, name).paginatedEntitiesOfType(entityType, { page, pageSize })
```

Comment:
It looks like this pagination code isn't covered by unit tests. Could you add a test case for it?

![image](https://user-images.githubusercontent.com/484484/195381846-931ea838-18d6-4cbf-8192-d95c5bd34fb6.png)